setup.py

```diff
@@ -37,7 +37,7 @@
     pass
 
 # Other requirements
-install_requires += ["scipy"]
+install_requires += ["scipy", "jaxtyping>=0.2.9", "typeguard~=2.13.3"]
```

Comment:
> @corwinjoy thanks for the detailed explanation! What you currently have works for me then :)

Cool! I think the short answer was that essentially Jaxtyping has a slightly different notion of broadcasting than LinearOperator. Essentially, Jaxtyping still wants the dimension to be there for broadcasting, but it may be 1. In contrast, LinearOperator will often squeeze / collapse dimensions.
Eventually, I think it would help to be more strict about how these routines are called since then we could have tighter contracts around the signatures. For now, I just tried to reflect the call variation that is present in the tests. For a similar perspective see the following paper which tries to automate this analysis:
https://arxiv.org/pdf/2112.09037.pdf